env/env_posix.cc

```diff
@@ -451,6 +451,47 @@ class PosixEnv : public Env {
     return Status::OK();
   }
 
+  virtual Status NewMemoryMappedFileBuffer(
```

Comment:
Right then we can propagate `munmap` errors, which is good (though not sure if they really happen)

Reply:
Oops looks like I misunderstood here. It's a bit hard to reuse in a clean way. It'd be good if we can think of a way but so far I cannot. The problems are:

- PosixMmapFile is a WritableFile so we'd have to either add the function for returning base to WritableFile, or static_cast down to PosixMmapFile in db_stress which would prevent it from running on windows.
- Also since PosixMmapFile is a WritableFile, it can expand its mmap region as the write progresses by calling MapNewRegion internally. We would need a way to invalidate raw buffer pointers.

Reply:
Good point. I can't think of a clean way to do this either. The WritableFile abstraction makes it hard to expose any mmap semantics.